### PR TITLE
Implemented checksum calculations in SW and HW offloading

### PR DESCRIPTION
**Pull request by mbulatova**
Support IPv4, TCP, UDP checksums

### CODE REVIEW

File: low/low.c [hunk 1]
```diff
@@ -133,6 +142,10 @@ void recv(uint8_t port, uint16_t queue, struct rte_ring *out_ring, uint8_t coreI
 		// Get RX packets from port
 		const uint16_t rx_pkts_number = rte_eth_rx_burst(port, queue, bufs, BURST_SIZE);
 
+        for (i = 0; i < rx_pkts_number; i++) {
+            printf("Recv packet flags = 0x%llx, packet type = 0x%x, tx_offload = 0x%016llx\n", bufs[i]->ol_flags, bufs[i]->packet_type, bufs[i]->tx_offload);
+        }
+
```
Comment (gshimansky) [inline low/low.c]: Please remove or comment debug print

File: packet/swcksum.go [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+	}
+
+	if length&1 != 0 {
+		sum += uint32(*(*byte)(unsafe.Pointer(uptr + uintptr(length-1)))) << 8
+	}
+
+	return sum
+}
+
+func CalculatePseudoHdrIPv4TCPCksum(hdr *IPv4Hdr) uint16 {
```
Comment (gshimansky) [inline packet/swcksum.go]: We need to add documentation for public functions in this file, those that start with capital letter.

File: test/stability/test_cksum/part2.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+	if ptr.F2 != 0 {
+		fmt.Printf("Bad data found in the packet: %x\n", ptr.F2)
+		println("TEST FAILED")
+		return
+	}
+
+	ptr.F2 = ptr.F1
+
+	if hwol {
+		packet.SetHWCksumOLFlags(pkt)
```
Comment (gshimansky) [inline test/stability/test_cksum/part2.go]: Is it necessary to calculate pseudo header checksum here too, like it is done in part1?

**Comment by ifilippov** [comment]
Looks good to me.

**Pull request merged**